src/guides/anatomy-of-an-http-transaction.md

```diff
@@ -0,0 +1,410 @@
+[`ReadableStream`][] interface. This stream can be listened to or piped elswhere
+just like any other stream. We can grab the data right out of the stream by
+listening to the stream's `'data'` and `'end'` events.
+
+The chunk emitted in each `'data'` event is a [`Buffer`][], but if you know that
+you will only be dealing with UTF-8 strings and no binary data, it makes sense
+to let JavaScript coerce the chunks into strings.
+
+```javascript
+var body = '';
+request.on('data', function(chunk) {
+  body += chunk;
+}).on('end', function() {
+  // at this point, `body` has the entire request body stored in it
+});
+```
+
+> **Note:** This may seem a tad tedious, and in many cases, it is. Luckily,
+there are modules like [`body`][] on [`npm`][] which can help hide away some of
+this logic. It's important to have a good understanding of what's going on
+before going down that road, and that's why you're here!
+
+## A Quick Thing About Errors
+
+Since the `request` object is a [`ReadableStream`][], it's also an
+[`EventEmitter`][]. As such, it behaves in a slightly different way when an
+error happens.
+
+An error in the `request` stream presents itself by emitting an `'error'` event
+on the stream. **If you don't have a listener for that event, the error will be
+*thrown*, which could crash your Node.js program.** You should therefore add an
+`'error'` listener on your request streams, even if you just log it and
+continue on your way. (Though it's probably best to send some kind of HTTP error
+response. More on that later.)
+
+```javascript
+request.on('error', function(err) {
+  // This prints the error message and stack trace to `stderr`.
+  console.error(err.stack);
+});
+```
+
+There are other ways of [handling these errors][] such as using [`domains`][] or
+other abstractions and tools, but always be aware that errors can and do happen,
+and you're going to have to deal with them.
+
+## What We've Got so Far
+
+At this point, we've covered creating a server, and grabbing the method, URL,
+headers and body out of requests. When we put that all together, it might look
+something like this:
+
+```javascript
+var http = require('http');
+
+http.createServer(function(request, response) {
+  var headers = request.headers;
+  var method = request.method;
+  var url = request.url;
+  var body = '';
+  request.on('error', function(err) {
+    console.error(err);
+  }).on('data', function(chunk) {
+    body += chunk;
+  }).on('end', function() {
+    // At this point, we have the headers, method, url and body, and can now
+    // do whatever we need to in order to respond to this request.
+  });
+}).listen(80); // Activates this server, listening on port 80.
+```
+
+So far we haven't touched on the `response` object at all, which is an instance
+of [`ServerResponse`][], which is a [`WriteableStream`][]. It contains many
+useful methods for sending data back to the client. We'll cover that next.
```

Comment:
What do you think about noting what happens if we run this code and try to hit it with a browser? It seems like a good setup to explain that, in Node, if you don't explicitly do _something_ with the response, Node won't close it or respond automatically or any other magic.